Replaced .format{}/% with f-strings in core/tools/datetime.py

JMBurley opened the pull request:
Modified to python3 format strings
ref: https://github.com/pandas-dev/pandas/issues/29547


- [x] contributes to fstring corrections in  #29547
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
~~whatsnew entry~~ (will not add now. I think the full set of changes in #29547 should get a single update entry rather than a piecemeal one)

JMBurley committed:
Replaced .format{}/% with f-strings

Modifeid to python3 format strings
ref: https://github.com/pandas-dev/pandas/issues/29547

JMBurley commented:
suggest labels clean & style for this pull request, & part of the 1.0 milestone.

JMBurley committed:
Update from black auot-formatter

`black pandas` altered the code formatting

JMBurley committed:
Revert "Update from black auto-formatter"

This reverts commit 63fd66d2fead9179b2d346e355896a11d74a16ac.

JMBurley committed:
Update from black auto-formatter to datetimes.py

`black pandas` auto-formatting update.  Changes line breaks due to f-strings being more concise.  Some of the changes were stupid and I have taken out the extra `" "` on single line `raise` statements.

jbrockmendel commented on pandas/core/tools/datetimes.py:
```diff
@@ -485,27 +484,20 @@ def _adjust_to_origin(arg, origin, unit):
-                    arg=arg, origin=origin
-                )
+                f"'{arg}' is not compatible with origin='{origin}'; "
+                "it must be numeric with a unit specified "
```
pls remove trailing space

JMBurley replied:
Done, see updated commit

jbrockmendel commented on pandas/core/tools/datetimes.py:
```diff
@@ -810,8 +802,8 @@ def f(value):
             "to assemble mappings requires at least that "
-            "[year, month, day] be specified: [{required}] "
-            "is missing".format(required=",".join(req))
+            f"[year, month, day] be specified: [{','.join(req)}] "
```
pls do the join in a separate line

JMBurley replied:
Done, see updated commit

JMBurley committed:
Updated in response to comments

remove trailing whitespace, join() not performed inline in f-strings.

update passes `black pandas` test

alimcmaster1 reviewed (approved):
LGTM

jreback commented:
thanks @JMBurley
